pipelines/toast_ground_sim.py

```diff
@@ -547,415 +162,173 @@ def parse_arguments(comm):
+        timer.report("Loading focalplanes")
     return detweights
 
 
 @function_timer
-def get_analytic_noise(args, comm, focalplane):
-    """Create a TOAST noise object.
-
-    Create a noise object from the 1/f noise parameters contained in the
-    focalplane database.
-
-    """
-    tmr = Timer()
-    tmr.start()
-    detectors = sorted(focalplane.keys())
-    fmin = {}
-    fknee = {}
-    alpha = {}
-    NET = {}
-    rates = {}
-    for d in detectors:
-        rates[d] = args.samplerate
-        fmin[d] = focalplane[d]["fmin"]
-        fknee[d] = focalplane[d]["fknee"]
-        alpha[d] = focalplane[d]["alpha"]
-        NET[d] = focalplane[d]["NET"]
-    nse = AnalyticNoise(
-        rate=rates, fmin=fmin, detectors=detectors, fknee=fknee, alpha=alpha, NET=NET
-    )
-    tmr.stop()
-    if comm.world_rank == 0:
-        tmr.report("Creating noise model")
-    return nse
-
-
-@function_timer
-def get_breaks(comm, all_ces, nces, args):
-    """ List operational day limits in the list of CES:s.
-
-    """
-    breaks = []
-    if args.skip_daymaps:
-        return breaks
-    do_break = False
-    for i in range(nces - 1):
-        # If current and next CES are on different days, insert a break
-        tz = args.timezone / 24
-        start1 = all_ces[i][3]  # MJD start
-        start2 = all_ces[i + 1][3]  # MJD start
-        scan1 = all_ces[i][4]
-        scan2 = all_ces[i + 1][4]
-        if scan1 != scan2 and do_break:
-            breaks.append(nces + i + 1)
-            do_break = False
-            continue
-        day1 = int(start1 + tz)
-        day2 = int(start2 + tz)
-        if day1 != day2:
-            if scan1 == scan2:
-                # We want an entire CES, even if it crosses the day bound.
-                # Wait until the scan number changes.
-                do_break = True
-            else:
-                breaks.append(nces + i + 1)
-
-    nbreak = len(breaks)
-    if nbreak < comm.ngroups - 1:
-        if comm.world_rank == 0:
-            print(
-                "WARNING: there are more process groups than observing days. "
-                "Will try distributing by observation.",
-                flush=True,
-            )
-        breaks = []
-        for i in range(nces - 1):
-            scan1 = all_ces[i][4]
-            scan2 = all_ces[i + 1][4]
-            if scan1 != scan2:
-                breaks.append(nces + i + 1)
-        nbreak = len(breaks)
-
-    if nbreak != comm.ngroups - 1:
-        raise RuntimeError(
-            "Number of observing days ({}) does not match number of process "
-            "groups ({}).".format(nbreak + 1, comm.ngroups)
-        )
-    return breaks
-
-
-@function_timer
-def create_observation(args, comm, all_ces_tot, ices, noise):
+def create_observation(args, comm, all_ces_tot, ices, noise, verbose=True):
     """ Create a TOAST observation.
 
     Create an observation for the CES scan defined by all_ces_tot[ices].
 
     """
```

Comment:
should we have also a description of all the arguments in the docstring?

Reply:
I completely reworked the Schedule, CES, Telescope, Site, Focalplane mess. It had evolved over a long time. Much cleaner now, thanks!